js/adapt-contrib-xapi.js

```diff
@@ -64,1568 +61,1516 @@ define([
+    this.isTerminated = true;
+  }
 
-              val = 'http://' + val;
-            }
-          }
+  /**
+  * Check Wrapper to see if all parameters needed are set.
+  */
+  checkWrapperConfig() {
+    if (this.xapiWrapper.lrs.endpoint && this.xapiWrapper.lrs.actor
+    && this.xapiWrapper.lrs.auth && this.xapiWrapper.lrs.activity_id ) {
+      return true;
+    }
+    return false;
+  }
 
-          newConfig[key] = val;
-        }
-      }, this);
+  /**
+   * Attempt to extract endpoint, user and password from the config.json.
+   */
+  setWrapperConfig() {
+    const keys = ['endpoint', 'user', 'password'];
+    const newConfig = {};
+
+    keys.forEach(key => {
+      let val = this.getConfig('_' + key);
 
-      if (!_.isEmpty(newConfig)) {
-        this.xapiWrapper.changeConfig(newConfig);
+      if (val) {
+        // Note: xAPI wrapper requires a trailing slash and protocol to be present
+        if (key === 'endpoint') {
+          val = val.replace(/\/?$/, '/');
 
-        if (!this.xapiWrapper.testConfig()) {
-          throw new Error('Incorrect xAPI configuration detected');
+          if (!/^https?:\/\//i.test(val)) {
+            Adapt.log.warn('adapt-contrib-xapi: "_endpoint" value is missing protocol (defaulting to http://)');
+
+            val = 'http://' + val;
+          }
         }
+
+        newConfig[key] = val;
       }
-    },
-
-    /**
-     * Gets the URL the course is currently running on.
-     * @return {string} The URL to the current course.
-     */
-    getBaseUrl: function() {
-      var url = window.location.origin + window.location.pathname;
-
-      Adapt.log.info('adapt-contrib-xapi: Using detected URL (' + url + ') as ActivityID');
-
-      return url;
-    },
-
-    getAttemptDuration: function() {
-      return this.startAttemptDuration + this.getSessionDuration();
-    },
-
-    getSessionDuration: function() {
-      return Math.abs((new Date()) - this.startTimeStamp);
-    },
-
-    /**
-     * Converts milliseconds to an ISO8601 duration
-     * @param {int} inputMilliseconds - Duration in milliseconds
-     * @return {string} - Duration in ISO8601 format
-     */
-    convertMillisecondsToISO8601Duration: function(inputMilliseconds) {
-      var hours;
-      var minutes;
-      var seconds;
-      var i_inputMilliseconds = parseInt(inputMilliseconds, 10);
-      var i_inputCentiseconds;
-      var inputIsNegative = '';
-      var rtnStr = '';
-
-      // Round to nearest 0.01 seconds.
-      i_inputCentiseconds = Math.round(i_inputMilliseconds / 10);
-
-      if (i_inputCentiseconds < 0) {
-        inputIsNegative = '-';
-        i_inputCentiseconds = i_inputCentiseconds * -1;
-      }
+    });
 
-      hours = parseInt(((i_inputCentiseconds) / 360000), 10);
-      minutes = parseInt((((i_inputCentiseconds) % 360000) / 6000), 10);
-      seconds = (((i_inputCentiseconds) % 360000) % 6000) / 100;
+    if (!newConfig.length > 0) {
```

Comment:
Drop the !
We should get into this if statement when newConfig **does** have length